data/routers/postDb-router.js

```diff
@@ -0,0 +1,135 @@
+const router = require('express').Router();
+
+const postDb = require('../helpers/postDb');
+const userDb = require('../helpers/userDb');
+
+router.post('/', (req, res) => {
+  const newPost = req.body;
+  if (newPost.text && newPost.user_id) {
+    userDb.getById(newPost.user_id).then(user => {
+      if (user) {
+        postDb
+          .insert(newPost)
+          .then(post => {
+            res.status(201).json(post);
+          })
+          .catch(err => {
+            res.status(500).json({
+              error: 'There was an error while saving the post to the database'
+            });
+          });
+      } else {
+        res
+          .status(404)
+          .json({ error: 'The user with the specified ID does not exist.' });
+      }
+    });
+  } else {
+    res.status(400).json({
+      error: 'Please provide a valid user_id and text for the post.'
+    });
+  }
+});
+
+router.get('/', (req, res) => {
+  postDb
+    .get()
+    .then(posts => {
+      res.status(200).json(posts);
+    })
+    .catch(err => {
+      res
+        .status(500)
+        .json({ error: 'The posts information could not be retrieved.' });
+    });
+});
+
+router.get('/:id', (req, res) => {
+  postDb
+    .getById(req.params.id)
+    .then(post => {
+      if (post) {
+        res.status(200).json(post);
+      } else {
+        res
+          .status(404)
+          .json({ error: 'The post with the specified ID does not exist.' });
+      }
+    })
+    .catch(err => {
+      res
+        .status(500)
+        .json({ error: 'The post information could not be retrieved.' });
+    });
+});
+
+// Since this function returns posts, I thought it belonged here in postDb-router.js rather than userDb-router.js
```

Comment:
this could be argued both ways honestly, so it's fine so long as you have something in a comment/readme file for frontend devs to know what choice you made and why